src/classes/class-cachemanager.php

```diff
@@ -0,0 +1,305 @@
+					wp_die(
+						esc_html__( 'Your request does not seem to be a valid one.', 'woocart-defaults' )
+					);
+				}
+
+				// Show notice after cache is flushed.
+				$action = sanitize_key( $_REQUEST['wc_cache'] );
+
+				if ( 'done' === $action ) {
+					add_action( 'admin_notices', [ &$this, 'show_notices' ] );
+				} elseif ( 'flush' === $action ) {
+					// Check for nonce.
+					check_admin_referer( 'wc_cache_nonce' );
+
+					// Flush cache.
+					$this->flush_cache();
+
+					// Redirect after the cache is flushed.
+					wp_redirect(
+						esc_url_raw(
+							add_query_arg(
+								[ 'wc_cache' => 'done' ]
+							)
+						)
+					);
+				}
+			}
+		}
+
+		/**
+		 * Hack into the ajax hook for theme and plugin editor.
+		 *
+		 * @codeCoverageIgnore
+		 */
+		public function inject_cache() {
+			ob_start( 'after_inject_cache' );
+		}
+
+		/**
+		 * This hook runs after we the action has been initiated.
+		 *
+		 * @codeCoverageIgnore
+		 */
+		public function after_inject_cache( $buffer ) {
+			// We empty cache over here.
+			$this->flush_cache();
+
+			return $buffer;
+		}
+
+		/**
+		 * Flush cache (OPcache, Redis object cache, and FCGI cache).
+		 */
+		protected function flush_cache() {
+			// OPcache.
+			$this->flush_opcache();
+
+			// Flush Redis cache.
+			$this->flush_redis_cache();
+
+			// Flush FCGI cache.
+			$this->flush_fcgi_cache( $this->fcgi_path );
+		}
+
+		/**
+		 * Displays notice after the cache is flushed.
+		 *
+		 * @codeCoverageIgnore
+		 */
+		public function show_notices() {
+			?>
+	  <div class="notice notice-success is-dismissible">
+		<p><strong><?php esc_html_e( 'Cache has been flushed successfully.', 'woocart-defaults' ); ?></strong></p>
+		<button type="button" class="notice-dismiss">
+		  <span class="screen-reader-text"><?php esc_html_e( 'Dismiss this notice.', 'woocart-defaults' ); ?></span>
+		</button>
+	  </div>
+			<?php
+		}
+
+		/**
+		 * Flush OPcache.
+		 *
+		 * @access protected
+		 */
+		protected function flush_opcache() {
+			// Flush OPcache.
+			opcache_reset();
+		}
+
+		/**
+		 * Flush Redis cache.
+		 *
+		 * @codeCoverageIgnore
+		 */
+		protected function flush_redis_cache() {
+			// Flush WordPress cache object.
+			wp_cache_flush();
+
+			// Purge redis keys.
+			// Connect to Redis instance.
+			$this->redis_connect();
+
+			// Continue if the connection was successful.
+			if ( $this->connected ) {
+				// Find keys matching `cache:*`.
+				$pattern = 'cache:*';
+
+				foreach ( new Iterator\Keyspace( $this->redis, $pattern ) as $key ) {
+					$this->redis->del( $key );
+				}
+			}
+		}
+
+		/**
+		 * Flush FCGI cache.
+		 */
+		protected function flush_fcgi_cache( $directory ) {
+			// Cache location.
+			$scan = new \RecursiveDirectoryIterator( $directory, \RecursiveDirectoryIterator::SKIP_DOTS );
+
+			// Scan directory for files.
+			$files = new \RecursiveIteratorIterator( $scan, \RecursiveIteratorIterator::CHILD_FIRST );
+
+			// Files counter.
+			$counter = 0;
```

Comment:
```suggestion
			$deleted = false;
```
I see you don't actually need a counter, but only a indication that files were deleted.